### PR TITLE
Interval _contains returns False

### PR DESCRIPTION
**Pull request by kshitij10496**
- [x] fixes issue #7805 
- [x] added tests

Ping @hargup @smichr @aktech

### CODE REVIEW

**Comment by Arafatk** [comment]
@kshitij10496 line 953 of test_sets.py contains a trailing whitespace.

**Comment by kshitij10496** [comment]
Thanks @Arafatk 
I have made the necessary changes

File: sympy/sets/tests/test_sets.py [hunk 1]
```diff
@@ -941,3 +942,13 @@ def test_issue_10113():
 def test_issue_10248():
     assert list(Intersection(S.Reals, FiniteSet(x))) == [
         And(x < oo, x > -oo)]
+
+
+def test_set_in_interval():
+    x = Symbol('x', real=True)
+    assert x in Interval(x, x+4)
+    assert Interval(1, 2) not in Interval(0, 5)
```
Comment (smichr) [inline sympy/sets/tests/test_sets.py]: This sure seems counterintuitive -- so we can have an intersection of two intervals but still say that the smaller one is not contained in the bigger one? If an integer can be in an Interval, what's wrong with seeing if a set (FiniteSet) of integers is in the Interval...or *all* numbers in a range (Interval) is in the Interval?

Comment (kshitij10496) [inline sympy/sets/tests/test_sets.py]: @smichr I raised a topic along this line [here](https://github.com/sympy/sympy/issues/7805) before opening a PR.
Even I am of the mind that atleast a FiniteSet can be `in` an Interval.

Comment (asmeurer) [inline sympy/sets/tests/test_sets.py]: "in" means element of, not subset of.

Comment (kshitij10496) [inline sympy/sets/tests/test_sets.py]: Okay.
I thought that by `in`, maybe we should check if a set is a subset of the other.
But now, I get the point across and I think we should check for FiniteSet of integers in the Interval too.

Comment (smichr) [inline sympy/sets/tests/test_sets.py]: > "in" means element of, not subset of.

OK, so thinking of a set of items, I can see that for preciseness reasons we would want `Interval(2,3) in FiniteSet(Interval(1,4), foo)` to return False -- it's literal in the sense that *that* Interval is not in the set just as `'a' in ['abc', 'foo']` would be false while `any('a' in I for I in ['abc', 'foo'])` would be true.

File: sympy/sets/sets.py [hunk 2]
```diff
@@ -964,6 +964,9 @@ def _boundary(self):
         return FiniteSet(self.start, self.end)
 
     def _contains(self, other):
+        if isinstance(other, Set):
+            return False
```
Comment (smichr) [inline sympy/sets/sets.py]: False -> false (the sympified version)

File: sympy/sets/sets.py [hunk 3]
```diff
@@ -964,6 +964,9 @@ def _boundary(self):
         return FiniteSet(self.start, self.end)
 
     def _contains(self, other):
+        if isinstance(other, Set):
```
Comment (smichr) [inline sympy/sets/sets.py]: I think this is too restrictive. *Anything* that won't compare in a Boolean should return false, perhaps? So everything after the test `other.is_real...` could be wrapped in a `try/except` to catch a bad boolean arg. But...Booleans aren't detecting bad args well enough:

This is ok:

```
>>> I<2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\core\expr.py", line 299, in __lt__
    raise TypeError("Invalid comparison of complex %s" % me)
TypeError: Invalid comparison of complex I
```

This isn't:

```
>>> Interval(1,2)<3
False
>>> FiniteSet(2)<3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\sets\sets.py", line 1996, in __lt__
    return self.is_proper_subset(other)
  File "sympy\sets\sets.py", line 361, in is_proper_subset
    raise ValueError("Unknown argument '%s'" % other)
ValueError: Unknown argument '3'
```

So perhaps you should just return False if other is not an Expr:

```
>>> isinstance(S(2), Expr)
True
>>> isinstance(x, Expr)
True
```

**Comment by kshitij10496** [comment]
@smichr Shouldn't we also add support for a set of integers in an `Interval` too ?

```
In[ ]: FiniteSet(1, 2) in Interval(0, 5)
Out[ ]: True
```

With the current implementation, this test is failing.

**Comment by asmeurer** [comment]
No. in means element of, not subset of. The elements of Interval(0, 5) are numbers, not sets. The current behavior is correct.

**Comment by smichr** [comment]
Looks like this needs a rebase and then let's commit this.

**Pull request closed without merging**